src/Microsoft.DotNet.Interactive.PowerShell/AzureShellConnection.cs

```diff
@@ -599,35 +698,47 @@ private async Task GetDeviceCode()
                     resourceUri: resourceUri,
                     body: body,
                     contentType: "application/x-www-form-urlencoded",
-                    method: HttpMethod.Post,
-                    ignoreError: true
+                    token: null,
+                    method: HttpMethod.Post
                 );
 
-                var authResponsePending = JsonSerializer.Deserialize<AuthResponsePending>(response);
-                if (authResponsePending.error == null)
+                if (response.IsSuccessStatusCode)
                 {
-                    var authResponse = JsonSerializer.Deserialize<AuthResponse>(response);
+                    // Authentication was successful.
+                    bytes = await response.Content.ReadAsByteArrayAsync();
+                    var authResponse = JsonSerializer.Deserialize<AuthResponse>(bytes);
+
                     _cachedAccessToken = authResponse.access_token;
                     _cachedRefreshToken = authResponse.refresh_token;
-                    return;
+
+                    break;
+                }
+
+                if (response.StatusCode != HttpStatusCode.BadRequest)
+                {
+                    // Unexpected request failure.
+                    await ThrowForFailedRequest(response, "Failed to poll for authentication. {0}.");
                 }
 
-                if (!authResponsePending.error.Equals("authorization_pending"))
+                // Status code was 'Bad Request'. It's possible we are still pending on authentication.
+                bytes = await response.Content.ReadAsByteArrayAsync();
+                var authResponsePending = JsonSerializer.Deserialize<AuthResponsePending>(bytes);
+
+                if (!authResponsePending.error.Equals("authorization_pending", StringComparison.Ordinal))
                 {
-                    throw new Exception($"Authentication failed: {authResponsePending.error_description}");
+                    throw new HttpRequestException($"Authentication failed: {authResponsePending.error_description}");
                 }
 
                 Thread.Sleep(deviceCode.interval * 1000);
```

Comment:
Would it be better to do `Task.Wait` here since it's already returning a `Task`?

Reply:
Since the method is async, we can do `await Task.Delay(deviceCode.interval * 1000)`.

Reply:
@TylerLeonhardt We are polling on the authentication result here, so there is no task to wait on.
@jonsequitur Good point, will replace with `Task.Delay`.

Reply:
Sorry I meant delay 😅